csv2/source.go

```diff
@@ -0,0 +1,103 @@
+package csv2
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"strings"
+
+	"github.com/pilosa/pdk"
+	"github.com/pkg/errors"
+)
+
+type Source struct {
+	rs pdk.RawSource
+
+	cur    pdk.NamedReadCloser
+	scan   *bufio.Scanner
+	header []string
+	line   int
+}
+
+func NewSourceFromRawSource(rs pdk.RawSource) *Source {
+	return &Source{
+		rs: rs,
+	}
+}
+
+func (s *Source) Record() (record interface{}, err error) {
+	if s.cur == nil {
+		s.cur, err = s.rs.NextReader()
+		if err != nil {
+			return nil, err
+		}
+		// scan header line
+		s.scan = bufio.NewScanner(s.cur)
+		if s.scan.Scan() && s.scan.Err() == nil {
+			s.header = strings.Split(s.scan.Text(), ",")
+			if err := validateHeader(s.header); err != nil {
+				s.cur = nil
+				s.scan = nil
+				s.line = 0
+				return nil, errors.Wrap(err, "validating header")
+			}
+		}
+	}
+	scan := s.scan
+	for scan.Scan() {
+		s.line++
+		if err := scan.Err(); err != nil {
+			return nil, err
+		}
+		txt := scan.Text()
+		if strings.TrimSpace(txt) == "" {
+			continue // skip empty lines. TODO: add stats tracking
+		}
+		row := strings.Split(txt, ",")
+		recordMap, err := parseRecord(s.header, row)
+		if err != nil {
+			return nil, errors.Wrapf(err, "parsing")
+		}
+		// add file and line number under the comma header since that can't
+		// be a header from the csv file.
+		recordMap[","] = fmt.Sprintf("%s:line%d", s.cur.Name(), s.line)
```

Comment:
Is anything using this? I think I would prefer to nix this line unless we need it for something.

Also technically there can be a header called "," in CSV, it would just have to be escaped.